src/components/TeamMemberTasks/TeamMemberTasks.jsx

```diff
@@ -15,257 +19,223 @@ import { getUserProfile } from '../../actions/userProfile';
+        const finalData = [];
+        // const userNotifications = [];
+        // const taskNotificationPromises = [];
+        const allManagingTeams = [];
 
-        const toDate = moment().tz('America/Los_Angeles').endOf('week').subtract(0, 'weeks');
+        // const teamMembers = [];
 
-        memberTimeEntriesPromises.push(
-          httpService
-            .get(ENDPOINTS.TIME_ENTRIES_PERIOD(member._id, fromDate, toDate))
-            .catch((err) => {}),
-        );
-      });
-      Promise.all(memberTimeEntriesPromises).then((data) => {
-        // merge time entries into each user obj
-        for (let i = 0; i < uniqueMembers.length; i++) {
-          uniqueMembers[i] = {
-            ...uniqueMembers[i],
-            timeEntries: data[i].data,
-          };
-        }
+        console.log('before calls');
 
-        // fetch all tasks for each member
-        uniqueMembers.forEach((member) => {
-          teamMemberTasksPromises.push(
-            httpService.get(ENDPOINTS.TASKS_BY_USERID(member._id)).catch((err) => {
-              if (err.status !== 401) {
-                console.log(err);
-              }
-            }),
-          );
+        // fetch all team members for each team
+        managingTeams.forEach((team) => {
+            teamMembersPromises.push(httpService.get(ENDPOINTS.TEAM_MEMBERS(team._id)));
         });
-        Promise.all(teamMemberTasksPromises).then((data) => {
-          // merge assigned tasks into each user obj
-          for (let i = 0; i < uniqueMembers.length; i++) {
-            uniqueMembers[i] = {
-              ...uniqueMembers[i],
-              tasks: data[i].data,
+
+        Promise.all(teamMembersPromises).then((data) => {
+          console.log('team members', data);       
+          // teamMembers.push(data[0].data);
+          for (let i = 0; i < managingTeams.length; i++) {
+            allManagingTeams[i] = {
+              ...managingTeams[i],
+              members: data[i].data,
             };
+            allMembers = allMembers.concat(data[i].data);
           }
 
-          // fetch full user profile for each team member
+          console.log('after teams call');
+        
+          // fetch all time entries for current week for all members
+          const uniqueMembers = _.uniqBy(allMembers, '_id');
           uniqueMembers.forEach((member) => {
-            userProfilePromises.push(
-              httpService.get(ENDPOINTS.USER_PROFILE(member._id)).catch((err) => {}),
+            const fromDate = moment()
+              .tz('America/Los_Angeles')
+              .startOf('week')
+              .subtract(0, 'weeks');
+        
+            const toDate = moment()
+              .tz('America/Los_Angeles')
+              .endOf('week')
+              .subtract(0, 'weeks');
+        
+            memberTimeEntriesPromises.push(
+              httpService.get(ENDPOINTS.TIME_ENTRIES_PERIOD(member._id, fromDate, toDate)).catch((err) => { }),
```

Comment:
For Every Member we are calling this endpoint, we need an endpoint that takes the list of members and returns the data with just one call to mongo.  I mean why are we even getting this data in team member tasks we should just be getting aggregated data where the total hours are calculated for you.